File: C#/DSA/Easy/Trees/PostorderTraversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution4
{
    public IList<int> PostorderTraversal(TreeNode root)
    {
        IList<int> res = [];
        void Postorder(TreeNode curr)
        {
            if (curr is null) return;
            Postorder(curr.left);
            Postorder(curr.right);
            res.Add(curr.val);
        }
        Postorder(root);
        return res;
    }
}

File: JS/Medium/BinaryTree/postorderTraversal/postorderTraversal.js
/**145
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var postorderTraversal = function (root) {
  let res = [];
  let postOrder = (curr) => {
    if (!curr) return;
    postOrder(curr.left);
    postOrder(curr.right);
    res.push(curr.val);
  };
  postOrder(root);
  return res;
};
